reinforced_lib/rlib.py

```diff
@@ -367,40 +378,88 @@ def fit(self, agent_id: int = 0) -> None:
 
         raise NotImplementedError()
 
-    def save_agent_state(self, agent_id: int = 0, path: str = None) -> None:
+    def save(self, path: str = None) -> None:
         """
-        Saves selected agent instance to file in X format.
+        Saves the state of the experiment to a file in lz4 format. For each agent both the state and the
+        initialization parameters are saved. The environment extension is saved as well to fully reconstruct the
+        experiment.
 
         Parameters
         ----------
-        agent_id : int, default=0
-            The identifier of agent instance.
         path : str
-            Path to the output file.
+            Path to the checkpoint file. If ``.pkl.lz4`` extension not detected, it will be appended automaticly.
         """
 
-        if not path:
-            raise ValueError('No path is specified.')
-
-        # TODO implement objects saving and agree on a format
-
-        raise NotImplementedError()
-
-    def load_agent_state(self, path) -> int:
+        if path is None:
+            path = os.path.join(os.path.expanduser("~"), f"checkpoint{self._lz4_ext}")
```

Comment:
Co Ty na to, żeby jeśli użytkownik sam nie poda nazwy, to dodawać do niej timestamp? Wydaje mi się, że to jest fajne, bo wtedy przez przypadek użytkownik nie nadpisze sam sobie swojej pracy, wtedy nazwa mogłaby być w formacie `checkpoint-[date]-[time].pkl.lz4`. Można też ewentualnie dodać coś, bo zasugeruje że to plik z naszej biblioteki, np. `rlib-checkpoint...`.

Reply:
taaaak, podoba mi sie

Reply:
Zrobiłem tak jak zasugerowałeś, i jeszcze raz zmieniłem te domyślne ścieżki. Traz można główny katalog z *checkpoints* wyspecyfikować jako parametr `RLib()`, a metoda `save()` zwraca teraz ścieżkę gdzie ostatecznie zapisano ten plik.